### ISSUE TITLE
Crash when trying to select as default 

### ISSUE THREAD

**Issue by Netocon** [acszo/Redomi#25]
### Describe the bug

Simply open the app and click "ok" on the guide, the app will crash trying to open "opening links" section.

### Expected behavior

The opening links section should open.

### Device

Redmi Note 10 Pro (MIUI 12.5)

### Additional context

The issue seems to occur when the app tries to open the "Opening links" section in the app info on MIUI devices. This process isn’t redirected correctly, causing the app to crash. However, it’s still possible to get the app working by using the "Activity Launcher" app to access Android’s stock "Manage Apps" settings and then set the default app there. While this workaround is effective, it prevents users from choosing which services to use, as Redomi doesn’t allow them to bypass its guide.

**Comment by acszo** [comment]
Are you able to send any logs? I don't have devices runnning MIUI

**Comment by acszo** [comment]
btw isn't MIUI 12.5 based on Android 11? The guide dialog isn't supposed to show up on devices running lower than Android 12, which is weird

**Comment by Netocon** [comment]
> Are you able to send any logs? I don't have devices running MIUI

Unfortunately not, my device is not able to see ADB for some reason

**Comment by Netocon** [comment]
> btw isn't MIUI 12.5 based on Android 11? The guide dialog isn't supposed to show up on devices running lower than Android 12, which is weird

You are right, i forgot that my device got updated to MIUI 13, i'll update the issue

**Comment by acszo** [comment]
> > Are you able to send any logs? I don't have devices running MIUI
> 
> Unfortunately not, my device is not able to see ADB for some reason

I just tried on a Redmi Note 12 Pro with MIUI 13 (cloud device) but the bug doesn't happen 🤔

Maybe try something in here to fix ADB? 
https://stackoverflow.com/questions/31488295/adb-is-not-recognizing-my-redmi-note-device

Or maybe try using Shizuku if you're familiar with it
